### PR TITLE
Navigation2

### PR DESCRIPTION
**Pull request by dominik-git**
You should know to navigate in video by keyboards left and right and when you press enter then is executed function for the button.

### CODE REVIEW

File: src/components/video-player-container/parts/video-player-controls.js [hunk 1]
```diff
@@ -1,82 +1,61 @@
...
         </div>
         <div className="player-interface__navigation-wrapper">
-          <button className="player-interface__navigation-control" onClick={this.props.fastRewind}>
-            <img src={Rewind} alt="right" className="player-interface-navigation-icons" />
+          <button
+            className={`player-interface__navigation-control ${selectedPosition === 0 ? 'selected' : ' '}`}
+          >
+            <img
+              src={Rewind}
+              alt="right"
```
Comment (garySiladi) [inline src/components/video-player-container/parts/video-player-controls.js]: you mean `left`?

File: src/components/video-player-container/parts/video-player-controls.js [hunk 2]
```diff
@@ -1,82 +1,61 @@
...
-      <div className={videoControlsClassName}>
+      <div className="player-interface">
         <div className="player-interface__progress-bar-container" >
           <div className="player-interface__progress-bar" style={progressBarStyle} />
         </div>
         <div className="player-interface__navigation-wrapper">
-          <button className="player-interface__navigation-control" onClick={this.props.fastRewind}>
-            <img src={Rewind} alt="right" className="player-interface-navigation-icons" />
+          <button
+            className={`player-interface__navigation-control ${selectedPosition === 0 ? 'selected' : ' '}`}
```
Comment (garySiladi) [inline src/components/video-player-container/parts/video-player-controls.js]: write:
```
className={`player-interface__navigation-control${selectedPosition === 0 ? ' selected' : ''}`}
```

File: src/components/video-player-container/parts/video-player-controls.js [hunk 3]
```diff
@@ -1,82 +1,61 @@
...
           </button>
           <button
-            className="player-interface__navigation-control"
-            onClick={this.props.fastForward}
+            className={`player-interface__navigation-control ${selectedPosition === 2 ? 'selected' : ' '}`}
           >
-            <img src={Forward} alt="left" className="player-interface-navigation-icons" />
+            <img
+              src={Forward}
+              alt="left"
```
Comment (garySiladi) [inline src/components/video-player-container/parts/video-player-controls.js]: you mean `right`?

File: src/components/video-player-container/parts/video-player-controls.js [hunk 4]
```diff
@@ -1,82 +1,61 @@
...
+  props: VideoPlayerControlsProps;
   render() {
     const progressBarStyle = {
       width: `${this.props.progress}%`,
     };
-    const videoControlsClassName= classnames({
-      'player-interface': true,
-      selected: this.props.isControlSelected,
-    });
+    const selectedPosition = this.props.selectedPosition;
```
Comment (garySiladi) [inline src/components/video-player-container/parts/video-player-controls.js]:
```
const {
  selectedPosition,
} = this.props;
```

File: src/components/video-player-container/parts/video-player-controls.js [hunk 5]
```diff
@@ -1,82 +1,61 @@
...
+              src={Rewind}
+              alt="right"
+              className="player-interface-navigation-icons"
+              id="1"
+            />
           </button>
           <button
-            onClick={this.props.isVideoPlaying ? this.props.pauseVideo : this.props.playVideo}
-            className="player-interface__navigation-control"
+            className={`player-interface__navigation-control ${selectedPosition === 1 ? 'selected' : ' '}`}
```
Comment (garySiladi) [inline src/components/video-player-container/parts/video-player-controls.js]: as above

File: src/components/video-player-container/parts/video-player-controls.js [hunk 6]
```diff
@@ -1,82 +1,61 @@
...
+              src={this.props.isVideoPlaying ? Pause : Play}
+              alt="Play/Pause"
+              className="player-interface-navigation-icons"
+              id="2"
+            />
           </button>
           <button
-            className="player-interface__navigation-control"
-            onClick={this.props.fastForward}
+            className={`player-interface__navigation-control ${selectedPosition === 2 ? 'selected' : ' '}`}
```
Comment (garySiladi) [inline src/components/video-player-container/parts/video-player-controls.js]: as above

File: src/components/video-player-container/parts/video-player-controls.test.js [hunk 7]
```diff
@@ -1,68 +1,44 @@
...
-function connectEvent(event, type, wrapper) {
-  const changedEvent: Object = event;
-  changedEvent.key = type;
-  const app: Object = wrapper.instance();
-  app.handleKeyPress(event);
+test('Video is playing + selected posstition is 0', () => {
+  const tree : string = renderer.create(
+    <VideoPlayerControls
+       isVideoPlaying
+       selectedPosition={ 0 }
```
Comment (garySiladi) [inline src/components/video-player-container/parts/video-player-controls.test.js]: no spaces

File: src/components/video-player-container/parts/video-player-controls.test.js [hunk 8]
```diff
@@ -1,68 +1,44 @@
...
-    />);
-  const event = new Event('keyDown');
-  expect(app.props().isControlSelected).toEqual(true);
-  expect(app.props().isVideoPlaying).toEqual(false);
-  connectEvent(event, 'ArrowLeft', app);
-  connectEvent(event, 'ArrowRight', app);
-  connectEvent(event, 'Enter', app);
-  connectEvent(event, 'Space', app);
+       isVideoPlaying
+       selectedPosition={ 2 }
```
Comment (garySiladi) [inline src/components/video-player-container/parts/video-player-controls.test.js]: no space

File: src/components/video-player-container/parts/video-player-controls.test.js [hunk 9]
```diff
@@ -1,68 +1,44 @@
...
-      progress={0}
-    />);
-  const event = new Event('keyDown');
-  expect(app.props().isControlSelected).toEqual(true);
-  connectEvent(event, 'ArrowLeft', app);
-  connectEvent(event, 'ArrowRight', app);
-  connectEvent(event, 'Enter', app);
-  connectEvent(event, 'Space', app);
+       isVideoPlaying
+       selectedPosition={ 1 }
```
Comment (garySiladi) [inline src/components/video-player-container/parts/video-player-controls.test.js]: no space

File: src/components/video-player-container/parts/video-player-controls.test.js [hunk 10]
```diff
@@ -1,68 +1,44 @@
...
+       selectedPosition={ 0 }
+       progress={0}
+     />).toJSON();
+  expect(tree).toMatchSnapshot();
+});
 
-}
-test('videoPlayerControls navigation works', () => {
-  const app = mount(
+test('Video is playing + selected posstition is 1', () => {
```
Comment (garySiladi) [inline src/components/video-player-container/parts/video-player-controls.test.js]: position

File: src/components/video-player-container/parts/video-player-controls.test.js [hunk 11]
```diff
@@ -1,68 +1,44 @@
...
-import { mount } from 'enzyme';
+import renderer from 'react-test-renderer';
 import VideoPlayerControls from './video-player-controls';
 
-function connectEvent(event, type, wrapper) {
-  const changedEvent: Object = event;
-  changedEvent.key = type;
-  const app: Object = wrapper.instance();
-  app.handleKeyPress(event);
+test('Video is playing + selected posstition is 0', () => {
```
Comment (garySiladi) [inline src/components/video-player-container/parts/video-player-controls.test.js]: position

File: src/components/video-player-container/parts/video-player-controls.js [hunk 12]
```diff
@@ -1,82 +1,61 @@
 // @flow
 import React from 'react';
-import classnames from 'classnames';
+// import classnames from 'classnames';
```
Comment (garySiladi) [inline src/components/video-player-container/parts/video-player-controls.js]: remove comment

File: src/components/video-player-container/parts/video-player-controls.test.js [hunk 13]
```diff
@@ -1,68 +1,44 @@
...
-    />);
-  const event = new Event('keyDown');
-  expect(app.props().isControlSelected).toEqual(false);
-  connectEvent(event, 'ArrowLeft', app);
-  connectEvent(event, 'ArrowRight', app);
-  connectEvent(event, 'Enter', app);
-  expect(app.props().isVideoPlaying).toEqual(true);
-  connectEvent(event, 'Space', app);
+       isVideoPlaying={ false }
+       selectedPosition={ 0 }
```
Comment (garySiladi) [inline src/components/video-player-container/parts/video-player-controls.test.js]: no space

File: src/components/video-player-container/parts/video-player-controls.test.js [hunk 14]
```diff
@@ -1,68 +1,44 @@
...
-      progress={0}
-    />);
-  const event = new Event('keyDown');
-  expect(app.props().isControlSelected).toEqual(false);
-  connectEvent(event, 'ArrowLeft', app);
-  connectEvent(event, 'ArrowRight', app);
-  connectEvent(event, 'Enter', app);
-  expect(app.props().isVideoPlaying).toEqual(true);
-  connectEvent(event, 'Space', app);
+       isVideoPlaying={ false }
```
Comment (garySiladi) [inline src/components/video-player-container/parts/video-player-controls.test.js]: no space

File: src/components/video-player-container/parts/video-player.js [hunk 15]
```diff
@@ -66,38 +71,86 @@ class VideoPlayer extends React.Component {
...
   }
-  handleNavigationState(event: KeyboardEvent) {
+  handleKeyPress(event: KeyboardEvent) {
     switch (event.key) {
       case 'ArrowUp':
         (this: any).setState({
-          isControlSelected: false,
+          isBackButtonSelected: true,
+          isNavigationSelected: false,
+          selectedPosition: null,
```
Comment (garySiladi) [inline src/components/video-player-container/parts/video-player.js]: you don't have to change `selectedPosition`

File: src/components/video-player-container/parts/video-player.js [hunk 16]
```diff
@@ -66,38 +71,86 @@ class VideoPlayer extends React.Component {
...
         break;
+      case 'ArrowLeft':
+        this.moveLeft();
+        break;
+      case 'ArrowRight':
+        this.moveRight();
+        break;
       case 'Enter':
-        if (!(this: any).state.isControlSelected) {
+        if ((this: any).state.isBackButtonSelected) {
```
Comment (garySiladi) [inline src/components/video-player-container/parts/video-player.js]: you don't have to put `(this: any)` flow type to these things ... that is only for the prop `player`

Comment (garySiladi) [inline src/components/video-player-container/parts/video-player.js]: change it everywhere just to `this`

File: src/components/video-player-container/parts/video-player.js [hunk 17]
```diff
@@ -66,38 +71,86 @@ class VideoPlayer extends React.Component {
...
+          if ((this: any).state.selectedPosition === 2) {
+            this.handleFastForward();
+          }
         }
         break;
       case 'Backspace':
         browserHistory.goBack();
         break;
       default:
+
```
Comment (garySiladi) [inline src/components/video-player-container/parts/video-player.js]: unneeded line

File: src/components/video-player-container/parts/video-player-controls.js (hunk 12 above)
Comment (dominik-git) [inline src/components/video-player-container/parts/video-player-controls.js]: fixed

File: src/components/video-player-container/parts/video-player-controls.js (hunk 2 above)
Comment (dominik-git) [inline src/components/video-player-container/parts/video-player-controls.js]: fixed

File: src/components/video-player-container/parts/video-player-controls.js (hunk 1 above)
Comment (dominik-git) [inline src/components/video-player-container/parts/video-player-controls.js]: fixed

File: src/components/video-player-container/parts/video-player-controls.js (hunk 5 above)
Comment (dominik-git) [inline src/components/video-player-container/parts/video-player-controls.js]: fixed

File: src/components/video-player-container/parts/video-player-controls.js (hunk 6 above)
Comment (dominik-git) [inline src/components/video-player-container/parts/video-player-controls.js]: fixed

File: src/components/video-player-container/parts/video-player-controls.js (hunk 3 above)
Comment (dominik-git) [inline src/components/video-player-container/parts/video-player-controls.js]: fixed

File: src/components/video-player-container/parts/video-player-controls.test.js (hunk 11 above)
Comment (dominik-git) [inline src/components/video-player-container/parts/video-player-controls.test.js]: fixed

File: src/components/video-player-container/parts/video-player-controls.test.js (hunk 7 above)
Comment (dominik-git) [inline src/components/video-player-container/parts/video-player-controls.test.js]: fixed

File: src/components/video-player-container/parts/video-player-controls.test.js (hunk 10 above)
Comment (dominik-git) [inline src/components/video-player-container/parts/video-player-controls.test.js]: fixed

File: src/components/video-player-container/parts/video-player-controls.test.js (hunk 9 above)
Comment (dominik-git) [inline src/components/video-player-container/parts/video-player-controls.test.js]: fixed

File: src/components/video-player-container/parts/video-player-controls.test.js (hunk 8 above)
Comment (dominik-git) [inline src/components/video-player-container/parts/video-player-controls.test.js]: fixed

File: src/components/video-player-container/parts/video-player-controls.test.js (hunk 14 above)
Comment (dominik-git) [inline src/components/video-player-container/parts/video-player-controls.test.js]: fixed

File: src/components/video-player-container/parts/video-player-controls.test.js (hunk 13 above)
Comment (dominik-git) [inline src/components/video-player-container/parts/video-player-controls.test.js]: fixed

File: src/components/video-player-container/parts/video-player.js (hunk 15 above)
Comment (dominik-git) [inline src/components/video-player-container/parts/video-player.js]: fixed

File: src/components/video-player-container/parts/video-player.js (hunk 16 above)
Comment (dominik-git) [inline src/components/video-player-container/parts/video-player.js]: fixed

File: src/components/video-player-container/parts/video-player.js (hunk 17 above)
Comment (dominik-git) [inline src/components/video-player-container/parts/video-player.js]: fixed

File: src/components/video-player-container/parts/video-player-controls.css [hunk 18]
```diff
@@ -43,6 +43,10 @@
   width: 45%;
 }
 
+.hide {
```
Comment (garySiladi) [inline src/components/video-player-container/parts/video-player-controls.css]: this classname is not used

File: src/components/video-player-container/parts/video-player-controls.test.js [hunk 19]
```diff
@@ -1,68 +1,44 @@
...
 
-function connectEvent(event, type, wrapper) {
-  const changedEvent: Object = event;
-  changedEvent.key = type;
-  const app: Object = wrapper.instance();
-  app.handleKeyPress(event);
+test('Video is playing + selected postition is 0', () => {
+  const tree : string = renderer.create(
+    <VideoPlayerControls
+       isVideoPlaying
```
Comment (garySiladi) [inline src/components/video-player-container/parts/video-player-controls.test.js]: the lines 9-12 have one extra space that is not needed. This is true for all similar props of this file.

File: src/components/video-player-container/parts/video-player.js (hunk 15 above)
Comment (garySiladi) [inline src/components/video-player-container/parts/video-player.js]: ok my bad

File: src/components/video-player-container/parts/video-player-controls.css [hunk 20]
```diff
@@ -43,6 +43,10 @@
...
+  display: none;
+}
+
 .selected {
```
Comment (varholak-peter) [inline src/components/video-player-container/parts/video-player-controls.css]: This is not a good name for a class, consider BEM.

File: src/components/video-player-container/parts/video-player-controls.js [hunk 21]
```diff
@@ -1,82 +1,62 @@
...
-          } else {
-            this.props.playVideo();
-          }
-          break;
-        default:
-      }
-    }
-  }
-
+class VideoPlayerControls extends React.Component { // eslint-disable-line react/prefer-stateless-function, max-len
```
Comment (varholak-peter) [inline src/components/video-player-container/parts/video-player-controls.js]: This should be a stateless component.

File: src/components/video-player-container/parts/video-player-controls.js [hunk 22]
```diff
@@ -1,82 +1,62 @@
...
-      <div className={videoControlsClassName}>
+      <div className="player-interface">
         <div className="player-interface__progress-bar-container" >
           <div className="player-interface__progress-bar" style={progressBarStyle} />
         </div>
         <div className="player-interface__navigation-wrapper">
-          <button className="player-interface__navigation-control" onClick={this.props.fastRewind}>
-            <img src={Rewind} alt="right" className="player-interface-navigation-icons" />
+          <button
+            className={`player-interface__navigation-control${selectedPosition === 0 ? ' selected' : ''}`}
```
Comment (varholak-peter) [inline src/components/video-player-container/parts/video-player-controls.js]: We should use `classnames()` here

File: src/components/video-player-container/parts/video-player-controls.js [hunk 23]
```diff
@@ -1,82 +1,62 @@
...
+          >
+            <img
+              src={Rewind}
+              alt="left"
+              className="player-interface-navigation-icons"
```
Comment (varholak-peter) [inline src/components/video-player-container/parts/video-player-controls.js]: this is not BEM

File: src/components/video-player-container/parts/video-player-controls.js [hunk 24]
```diff
@@ -1,82 +1,62 @@
...
+              id="1"
+            />
           </button>
           <button
-            onClick={this.props.isVideoPlaying ? this.props.pauseVideo : this.props.playVideo}
-            className="player-interface__navigation-control"
+            className={`player-interface__navigation-control${selectedPosition === 1 ? ' selected' : ''}`}
```
Comment (varholak-peter) [inline src/components/video-player-container/parts/video-player-controls.js]: We should use `classnames()` here

File: src/components/video-player-container/parts/video-player-controls.js [hunk 25]
```diff
@@ -1,82 +1,62 @@
...
           >
-            <img src={this.props.isVideoPlaying ? Pause : Play} alt="Play/Pause" className="player-interface-navigation-icons" />
+            <img
+              src={this.props.isVideoPlaying ? Pause : Play}
+              alt="Play/Pause"
+              className="player-interface-navigation-icons"
```
Comment (varholak-peter) [inline src/components/video-player-container/parts/video-player-controls.js]: not BEM

File: src/components/video-player-container/parts/video-player-controls.js [hunk 26]
```diff
@@ -1,82 +1,62 @@
...
+              id="2"
+            />
           </button>
           <button
-            className="player-interface__navigation-control"
-            onClick={this.props.fastForward}
+            className={`player-interface__navigation-control${selectedPosition === 2 ? ' selected' : ''}`}
```
Comment (varholak-peter) [inline src/components/video-player-container/parts/video-player-controls.js]: We should use `classnames()` here

File: src/components/video-player-container/parts/video-player-controls.js [hunk 27]
```diff
@@ -1,82 +1,62 @@
...
           >
-            <img src={Forward} alt="left" className="player-interface-navigation-icons" />
+            <img
+              src={Forward}
+              alt="right"
+              className="player-interface-navigation-icons"
```
Comment (varholak-peter) [inline src/components/video-player-container/parts/video-player-controls.js]: not BEM

File: src/components/video-player-container/parts/video-player-controls.css (hunk 18 above)
Comment (dominik-git) [inline src/components/video-player-container/parts/video-player-controls.css]: fixed

File: src/components/video-player-container/parts/video-player-controls.css (hunk 20 above)
Comment (varholak-peter) [inline src/components/video-player-container/parts/video-player-controls.css]: This was not addressed.

File: src/components/video-player-container/parts/video-player-controls.js [hunk 28]
```diff
@@ -1,66 +1,76 @@
...
+    selected: selectedPosition === 0,
+  });
+  const playPauseButtonClassnames = classnames({
+    'player-interface__navigation-control': true,
+    selected: selectedPosition === 1,
+  });
+  const fastForwardButtonClassnames = classnames({
+    'player-interface__navigation-control': true,
+    selected: selectedPosition === 2,
+  });
```
Comment (varholak-peter) [inline src/components/video-player-container/parts/video-player-controls.js]: We should remake these 3 classes into a className generator function.

```
const createNavigationControlClassname = position => classnames({
  'player-interface__navigation-control': true,
  selected: selectedPosition === position,
})
...
<button className={createNavigationControlClassname(0)}>
<button className={createNavigationControlClassname(1)}>
<button className={createNavigationControlClassname(2)}>
```

**Pull request merged**